util/topgen/intermodule.py

```diff
@@ -69,6 +69,93 @@ def get_suffixes(ims: OrderedDict) -> (str, str):
     return ("_req", "_rsp")
 
 
+def add_intermodule_connection(obj, req_m, req_s, rsp_m, rsp_s):
+    """Add if doesn't exist the connection
+    """
+    req_key = "{}.{}".format(req_m, req_s)
+    rsp_key = "{}.{}".format(rsp_m, rsp_s)
+
+    connect = obj["inter_module"]["connect"]
+    if req_key in connect:
+        # check if rsp has data
+        if rsp_key in connect[req_key]:
+            return
+        req_key.append(rsp_key)
+        return
+
+    # req_key is not in connect:
+    # check if rsp_key
+    if rsp_key in connect:
+        # check if rsp has data
+        if req_key in connect[rsp_key]:
+            return
+        rsp_key.append(req_key)
+        return
+
+    # Add new key and connect
+    connect[req_key] = [rsp_key]
+
+
+def autoconnect(topcfg: OrderedDict):
+    """Matching the connection based on the naming rule
+    between {memory, module} <-> Xbar.
+    """
+
+    instances = topcfg["module"] + topcfg["memory"]
+
+    for xbar in topcfg["xbar"]:
+        ports = [x for x in xbar["nodes"] if x["type"] in ["host", "device"]]
+        for port in ports:
+            if port["xbar"]:
+                if port["type"] == "host":
+                    # Skip as device will add connection
+                    continue
+                add_intermodule_connection(obj=topcfg,
+                                           req_m=xbar["name"],
+                                           req_s="tl_%s" % port["name"],
+                                           rsp_m=port["name"],
+                                           rsp_s="tl_%s" % xbar["name"])
+
+            # check if name exists in ["module", "memory"]
+            ips = [x for x in instances if x["name"] == port["name"]]
+            if len(ips) == 1:
```

Comment:
should this be an error if length > 1?

Reply:
Thanks, revised slightly . New code has smaller indentations :)